ecl2df/common.py

```diff
@@ -114,13 +115,142 @@
 logger: logging.Logger = logging.getLogger(__name__)
 
 
-def write_dframe_stdout_file(
+def set_name_from_args(args):
+    """Make file name from input arguments
+
+    Args:
+        args (dict): input arguments
+
+    Returns:
+        str: name of file to
+    """
+
+    name, tagname, _ = get_names_from_args(args)
+    out_name = f"{name}--{tagname}.csv"
+    try:
+        if args["arrow"]:
+            try:
+                out_name = out_name.replace(".csv", ".arrow")
+            except AttributeError:
+                print("No replacement here today!")
+
+    except KeyError:
+        logger.debug("No arrow format to write")
+
+    return out_name
+
+
+def get_names_from_args(args):
+    """Get name components from input arguments, mainly name of ecl file
+
+    Args:
+        args (dict): input arguments
+
+    Returns:
+        str: the name to use when exporting
+    """
+    subcommand = args["subcommand"]
+
+    contents = {
+        "summary": "timeseries",
+    }
+
+    content = contents.get(subcommand, "property")
+
+    try:
+        file_name = Path(args["DATAFILE"])
+    except KeyError:
+        file_name = Path(args["PRTFILE"])
+
+    logger.debug("File name: %s", file_name)
+
+    tagname = subcommand
+
+    base_name = file_name.name
+    # if no suffix supplied
+    while base_name[-1].isdigit() or base_name.endswith("-"):
+        base_name = base_name[:-1]
+
+    name = re.sub(r"(-\d+)?\..*", "", base_name)
+    logger.debug("Name and tag " + name + "|" + tagname)
+
+    return name, tagname, content
+
+
+def write_dframe_to_file(
     dframe: Union[pd.DataFrame, pyarrow.Table],
     output: str,
```

Comment:
I don't like changing this arg from a well-defined string to a big and complex object (an argparse.Namespace), making it enormously harder to test. Can this be refactored?